File: Practica 6/Ejercicio 5/Imprime_pares.py
def imprime_pares_intervalo(num1, num2): # Define una función llamada "imprime_pares_intervalo"

    for i in range (num1, num2 + 1): # Crea un bucle con determinadas iteraciones hasta que se cumpla la condición, que es que recorra desde el "num1" hasta el "num2" (ambos incluidos)
        if (i % 2 == 0): # Crea un condicional if (si) que ejecuta el siguiente código si se cumple la condición, que es que el resto de dividir algún número entre "num1" y "num2" entre 2, su resto sea igual a 0
            print("\nLos números pares que hay entre",num1,"y",num2,"son:", i, end=" ") # Imprime el mensaje i veces con las variables que lo hayan cumplido y hace un salto de linea [end=" "]

def main(): # Define la función principal "main"
    
    num1 = int(input("\nEscriba un número: ")) # Imprime el mensaje en la consola [""], guarda los datos introducizos por el usuario [input] de tipo númerico [int] y lo guarda en la variable "num1"
    num2 = int(input("\nEscriba otro número: ")) # Imprime el mensaje en la consola [""], guarda los datos introducizos por el usuario [input] de tipo númerico [int] y lo guarda en la variable "num2"
    
    while (num1 > num2): # Crea un bucle con infinitas iteraciones hasta que la condición no se cumpla
        print("\nError, introduzca el primer número menor que el segundo") # Imprime el mensaje en la consola
        num1 = int(input("\nEscriba un número: ")) # Se reescribe el valor de la variable "num1" con el mismo proceso que antes
        num2 = int(input("\nEscriba otro número: ")) # Se reescribe el valor de la variable "num2" con el mismo proceso que antes
        
    imprime_pares_intervalo(num1, num2) # Llama a la función "imprime_pares_intervalo" y se ejecuta
    
main() # Llama a la función principal "main" y ejecuta el programa

File: Practicas/Practica 4/Ejercicio 5/ImprimePares.c
#include <stdio.h>

int imprime_pares_intervalo(int num1, int num2) {

    for (int i = num1; i <= num2; i++) {
        if (i % 2 == 0) {
            printf("\nLos números pares entre %d y %d son: ", num1, num2);
            printf("%d ", i);
        }
    }
    printf("\n");
}

int main() {

    int num1, num2;

    printf("\nPorfavor, introduzca un número: ");
    scanf("%d", &num1);

    printf("\nPorfavor, introduzca otro número: ");
    scanf("%d", &num2);

    while (num1 > num2) {
        printf("\nError, introduzca unos números validos");

        printf("\nVuelva a introducir un número valido: ");
        scanf("%d", &num1);

        printf("\nVuelva a introducir otro número valido: ");
        scanf("%d", &num2);
    }

    imprime_pares_intervalo(num1, num2);
    
    return 0;
}